Core/src/main/java/org/tribuo/util/MeanVarianceAccumulator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.tribuo.util;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * An accumulator for online calculation of the mean and variance of a
+ * stream of doubles.
+ * <p>
+ * Note this class is not thread safe.
+ */
+public final class MeanVarianceAccumulator implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private double max = Double.NEGATIVE_INFINITY;
+    private double min = Double.POSITIVE_INFINITY;
+
+    private double mean = 0.0;
+    private double sumSquares = 0.0;
+
+    private long count = 0;
+
+    /**
+     * Constructs an empty mean/variance accumulator.
+     */
+    public MeanVarianceAccumulator() {}
+
+    /**
+     * Constructs a mean/variance accumulator and observes the supplied array.
+     * @param values The array to operate on.
+     */
+    public MeanVarianceAccumulator(double[] values) {
+        observe(values);
+    }
+
+    /**
+     * Copy constructor.
+     * @param other The MeanVarianceAccumulator to copy.
+     */
+    public MeanVarianceAccumulator(MeanVarianceAccumulator other) {
+        this.max = other.max;
+        this.min = other.min;
+        this.mean = other.mean;
+        this.sumSquares = other.sumSquares;
+        this.count = other.count;
+    }
+
+    /**
+     * Resets this accumulator to the starting state.
+     */
+    public void reset() {
+        this.max = Double.NEGATIVE_INFINITY;
+        this.min = Double.POSITIVE_INFINITY;
+        this.mean = 0;
+        this.sumSquares = 0;
+        this.count = 0;
+    }
+
+    /**
+     * Observes a value, i.e. updates the mean, variance, max and min statistics.
```

Comment:
I get what you're going for here, but this comment is actually inaccurate. It doesn't update the variance, because that's computed when it's asked for.

Reply:
Fixed.